feat(charts): Allow custom environment config for fluentd daemon. [redux]

awmichel opened the pull request:
Corrected PR for #78.

See explanation in #79.

Sorry for the trouble, can someone also double check the output is correct now? I tested with the following values.yml:
```
daemon_environment:
  DROP_FLUENTD_LOGS: true
  FLUENTD_PLUGIN_1: fluent-plugin-aws-elasticsearch-service
```

Which yielded the following by calling `helm install --dry-run --debug charts/fluentd --values values.yml`:
```
SERVER: "192.168.64.4:32767"
CHART PATH: /Users/amichel/Development/GitHub/fluentd/charts/fluentd
NAME:   gone-salamander
TARGET NAMESPACE:   default
CHART:  fluentd <Will be populated by the ci before publishing the chart>
MANIFEST:
---
# Source: fluentd/templates/logger-fluentd-service-account.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: deis-logger-fluentd
  labels:
    heritage: deis

---
# Source: fluentd/templates/logger-fluentd-daemon.yaml
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: deis-logger-fluentd
  labels:
    heritage: deis
  annotations:
    component.deis.io/version: canary
spec:
  selector:
    matchLabels:
      app: deis-logger-fluentd
      heritage: deis
  template:
    metadata:
      name: deis-logger-fluentd
      labels:
        heritage: deis
        app: deis-logger-fluentd
    spec:
      serviceAccount: deis-logger-fluentd
      containers:
      - name: deis-logger-fluentd
        image: quay.io/deisci/fluentd:canary
        imagePullPolicy: Always
        env:
          - name: DROP_FLUENTD_LOGS
            value: true
          - name: FLUENTD_PLUGIN_1
            value: fluent-plugin-aws-elasticsearch-service
        volumeMounts:
        - name: varlog
          mountPath: /var/log
        - name: varlibdockercontainers
          mountPath: /var/lib/docker/containers
          readOnly: true
      volumes:
      - name: varlog
        hostPath:
          path: /var/log
      - name: varlibdockercontainers
        hostPath:
          path: /var/lib/docker/containers
```

awmichel committed:
Fix `env:` section formatting in output.

deis-admin commented:
Thanks for the contribution! Please ensure your commits follow our [style guide](http://docs.deis.io/en/latest/contributing/standards/#commit-style-guide). This code will be tested once a Deis maintainer reviews it.

kmala commented:
can you make the changes in a single PR instead of separate PR for revert. Also i would think `DROP_FLUENTD_LOGS` should be part of the daemon configuration and not part of the syslog section. @rimusz

kmala commented:
Jenkins, Add to whitelist.

rimusz commented:
yes, `DROP_FLUENTD_LOGS` should be part of the daemon configuration

awmichel committed:
Remove DROP_FLUENTD_LOGS from syslog config.

awmichel commented:
@kmala I closed the revert PR and rebased this one against master so it will cleanly merge. This now contains the fixes to #78.

@kmala && @rimusz I removed `DROP_FLUENTD_LOGS` from being added when using syslog values. This was a confusion from the discussion in #78.

bacongobbler commented:
thanks!
